component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/LabelDAO.java

```
@@ -0,0 +1,58 @@
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.exoplatform.social.core.jpa.storage.dao.jpa;
+
+import org.exoplatform.commons.persistence.impl.GenericDAOJPAImpl;
+import org.exoplatform.social.core.jpa.storage.entity.LabelEntity;
+import javax.persistence.TypedQuery;
+import java.util.ArrayList;
+import java.util.List;
+
+public class LabelDAO extends GenericDAOJPAImpl<LabelEntity, Long> {
+  public LabelEntity findLabelByObjectTypeAndObjectIdAndLang(String objectType, String objectId, String language) {
+    TypedQuery<LabelEntity> query = getEntityManager()
+            .createNamedQuery("SocLabelEntity.findLabelByObjectTypeAndObjectIdAndLang",
+                    LabelEntity.class)
+            .setParameter("objectType", objectType)
+            .setParameter("objectId", objectId)
+            .setParameter("language", language);
+    try {
+      return query.getSingleResult();
+    } catch (Exception e) {
+      return null;
+    }
+
+  }
+
+  public List<LabelEntity> findLabelByObjectTypeAndObjectId(String objectType, String objectId) {
+    TypedQuery<LabelEntity> query = getEntityManager()
+            .createNamedQuery("SocLabelEntity.findLabelByObjectTypeAndObjectId",
+                    LabelEntity.class)
+            .setParameter("objectType", objectType)
+            .setParameter("objectId", objectId);
+    try {
+      return query.getResultList();
+    } catch (Exception e) {
+      return new ArrayList<>();
```

please throw the error to make sure that the stored data is coherent, else it can result to a blocker bugs when swallowing general exceptions like this.

---

Done